quantuminspire/config_manager/config_manager.py

```diff
@@ -0,0 +1,253 @@
+        """Get the value of a setting and the source settings object that provides it.
+
+        This method searches through all configured settings objects in a defined
+        priority order (typically project → user) to determine the value for a given
+        key and which settings object it came from.
+
+        The resolution follows these rules:
+            1. Skip any settings object that does not have the requested attribute.
+            2. Return the first non-None value found, along with its source.
+            3. If all values are None but the attribute exists in at least one object,
+            return the last seen source and value (even if None).
+
+        Args:
+            key: The dot-separated setting key to retrieve.
+
+        Returns:
+            A tuple (source_name, value):
+                - source_name: The class name of the settings object that provided the value.
+                - value: The resolved value of the setting (can be None if all values are None).
+
+        Examples:
+            # Suppose project settings has `backend_type=None` and user settings has `backend_type=2`
+            source, value = config._get_source_and_value("backend_type")
+            # Returns ('UserSettings', 2)
+
+            # If the attribute exists only in project settings as None
+            source, value = config._get_source_and_value("project.id")
+            # Returns ('ProjectSettings', None)
+        """
+
+        last_value = None
+        last_source = None
+
+        for setting in self._settings_priority:
+            source_name = setting.__class__.__name__
+            try:
+                value = setting.get_value(key)
+            except AttributeError:
+                continue  # Attribute not on this settings object, try next
+
+            # Track last seen attribute (even if None)
+            last_value = value
+            last_source = source_name
+
+            # If value is not None, return immediately
+            if value is not None:
+                return last_source, last_value
+
+        # Fallback: attribute existed but all values were None
+        assert last_source is not None
+        return last_source, last_value
+
+    def get(self, key: str) -> Any:
+        """Retrieve the value of a configurable setting by key.
+
+        Args:
+            key: The dot-separated setting key to retrieve.
+
+        Returns:
+            The current value of the setting.
+
+        Raises:
+            ValueError: If the key is not a valid configurable setting.
+        """
+        if key not in self._configurable_keys:
+            raise ValueError(f"Key '{key}' is invalid")
+        _, value = self._get_source_and_value(key)
+        return value
+
+    def _get_setting_for_scope(
+        self, scope: ConfigScopes, is_global: bool
+    ) -> Tuple[BaseConfigSettings, Optional[BaseConfigSettings]]:
+        """Return the primary and secondary settings objects for a given scope.
```

Comment:
Why do you want to give primary and secondary? At some point, we will have CLI, project and user. Should we then update to a tuple of 3?